amiqus/models/step.py

```diff
@@ -17,10 +21,30 @@ class Step(models.Model):
 
     id = models.AutoField(primary_key=True)
 
+    amiqus_id = models.CharField(
+        "Amiqus ID",
+        max_length=40,
+        help_text=_(
+            "The id returned from the Amiqus API. These should be unique on "
+            "the record."
+        ),
+        null=True,
+    )
+
     record = models.ForeignKey(Record, on_delete=models.CASCADE, related_name="steps")
     amiqus_check = models.OneToOneField(
         Check, on_delete=models.CASCADE, null=True, blank=True, related_name="step"
     )
     form = models.OneToOneField(
         Form, on_delete=models.CASCADE, null=True, blank=True, related_name="step"
     )
+
+    def __str__(self) -> str:
+        return f"""Step number {self.id} on record {self.record.amiqus_id}
+            for {self.record.user.get_full_name()}
+            ({self.amiqus_check.check_type if self.amiqus_check else "form"})
+        """
```

Comment:
This `__str__` method causes a lot of table jumps, which can be quite inefficient when rendering a full list of steps in the Admin. Do we really require all of this info?